src/tests/unit/cpp_wrapper/test_cpp_wrapper.cpp

```diff
@@ -112,6 +113,439 @@ test_cpp_wrapper_insert_on_all_implementations(
 	test_cpp_wrapper_insert_get(tc, dict);
 	test_cpp_wrapper_insert_get_edge_cases(tc, dict);
 	delete dict;
+
+	dict = new OpenAddressFileHash<int, int>(key_type_numeric_signed, sizeof(int), sizeof(int), 50);
+	test_cpp_wrapper_insert_get(tc, dict);
+	test_cpp_wrapper_insert_get_edge_cases(tc, dict);
+	delete dict;
+}
+
+/**
+@brief	Tests an insertion and then attempts to delete the record.
+*/
+void
+test_cpp_wrapper_insert_delete(
+		planck_unit_test_t *tc,
+		Dictionary<int, int> *dict
+) {
+	for (int i = 0; i < 10; i++) {
+		dict->insert(i, i * 2);
+		PLANCK_UNIT_ASSERT_TRUE(tc, err_ok == dict->last_status.error);
+		PLANCK_UNIT_ASSERT_INT_ARE_EQUAL(tc, 1, dict->last_status.count);
+
+		dict->deleteRecord(i);
+		int ret_val = dict->get(i);
+
+		PLANCK_UNIT_ASSERT_TRUE(tc, err_ok != dict->last_status.error);
```

Comment:
Do a stricter check here and assert that `last_status.error` returns `err_item_not_found`.